internal/repository/sync.go

```diff
@@ -64,8 +68,33 @@ func Clone(ctx context.Context, remoteURL, dir, initialBranch string) (*Reposito
 		return nil, errors.Join(ErrCloningRepository, err)
 	}
 
+	sort.Slice(expectedRootKeys, func(i, j int) bool {
+		return expectedRootKeys[i].KeyID < expectedRootKeys[j].KeyID
+	})
+
 	repository := &Repository{r: r}
 
+	if len(expectedRootKeys) > 0 {
+		state, err := policy.LoadFirstState(ctx, r)
+		if err != nil {
+			return repository, errors.Join(ErrCloningRepository, err)
+		}
+		rootKeys, err := state.GetRootKeys()
+		if err != nil {
+			return repository, errors.Join(ErrCloningRepository, err)
+		}
+
+		slog.Debug("Verifying if root keys are expected root keys...")
```

Comment:
I'd move this even further up, right at the start of the if block. Loading all policy states to find the first one may take a moment, and it'd be good to know why.

Reply:
Moved